pylifx/networking.py

```diff
@@ -39,24 +43,61 @@
                 _AVAILABLE_INTERFACES[intf_name] = addr
                 break
 
+
 def get_interfaces():
     return _AVAILABLE_INTERFACES
 
+
 def get_interface(intf_name):
     if intf_name is None:
         return _AVAILABLE_INTERFACES.itervalues().next()
     else:
         return _AVAILABLE_INTERFACES[intf_name]
 
+
+def find_master_bulb(broadcast_address):
+    try:
+        broadcast_socket = socket(AF_INET, SOCK_DGRAM)
+        broadcast_socket.settimeout(3)
+        broadcast_socket.bind(('', _LFIX_BROADCAST_PORT))
+
+        sock = socket(AF_INET, SOCK_DGRAM)
+        sock.setsockopt(SOL_SOCKET, SO_REUSEADDR, 1)
+        sock.setsockopt(SOL_SOCKET, SO_BROADCAST, 1)
+
+        sock.sendto(
+            encode(
+                'getPanGateway',
+                protocol=_LIFX_PROTO_TOBULB,
+                bulb_addr=processMAC(_BLANK_MAC),
+                site_addr=processMAC(_BLANK_MAC),
+            ).bytes,
+            (broadcast_address, _LIFX_BROADCAST_PORT)
+        )
+
+        while True:
```

Comment:
This loop waits indefinitely for a response. Perhaps this should have a timeout?